values.tf

```diff
@@ -1,13 +1,26 @@
 locals {
-  values_default = yamlencode({
-    # add default values here
+  values_irsa = yamlencode({
+    "controller" : {
+      "serviceAccount" : {
+        "annotations" : {
+          "eks.amazonaws.com/role-arn" : aws_iam_role.this[0].arn
+        }
+      }
+    }
+    "server" : {
+      "serviceAccount" : {
+        "annotations" : {
+          "eks.amazonaws.com/role-arn" : aws_iam_role.this[0].arn
+        }
+      }
+    }
   })
 }
 
 data "utils_deep_merge_yaml" "values" {
   count = var.enabled ? 1 : 0
   input = compact([
-    local.values_default,
+    var.irsa_role_create ? local.values_irsa : yamlencode({}),
```

Comment:
Why do we use irsa role create condition differently than is standard in our modules? In standard case we use it like this:
```
locals {
  values = yamlencode({
    "controller" : {
      "serviceAccount" : {
        "annotations" : {
          "eks.amazonaws.com/role-arn" : local.irsa_role_create ? aws_iam_role.this[0].arn : ""
        }
      }
    }
    "server" : {
      "serviceAccount" : {
        "annotations" : {
          "eks.amazonaws.com/role-arn" : local.irsa_role_create ? aws_iam_role.this[0].arn : ""
        }
      }
    }
  })
}

data "utils_deep_merge_yaml" "values" {
  count = var.enabled ? 1 : 0
  input = compact([
    local.values,
    var.values
  ])
}
```

Ideally, we try to have things the same everywhere to avoid unnecessary differences that may lead to bugs.

Reply:
ok sure, I can align it. I did not check other addon how we do it there. I just created this by myself.

Reply:
@JaroVojtek there must be condition in role assignment with `local. irsa_role_create` otherwise there will be an error in some use cases

Reply:
values.tf in your comment is marked as outdated. when you open in PR file changed you can see update there.